feat: Add codes per publisher per codelist page to replace popup

Bjwebb opened the pull request:
https://github.com/IATI/IATI-Dashboard/issues/653

simon-20 commented on iati_dashboard/templates/codelist_publisher.html:
```diff
@@ -0,0 +1,89 @@
+<h2>Codelist values used for <code class="iati-code">{{ element }}</code></h2>
+<p class="dashboard-page-heading__lead">(for files published to version {{ major_version }}.xx of the standard)</p>
+<p style="float:right">(<a href="{{ stats_url }}/current/aggregated-publisher/{{ publisher.short_name }}/codelist_values.json">This page in JSON format</a>)</p>
+<p>Values should be on the <code class="iati-code"><a href="{% if major_version=='2' %}http://iatistandard.org/codelists/{% else %}https://iatistandard.org/105/codelists/{% endif %}{{ codelist_mapping[major_version].get(element) }}/">{{ codelist_mapping[major_version].get(element) }}</a></code> codelist.</p>
```
Should the version 2 links also be `https`?

Bjwebb replied:
I've fixed this for this template. This was copy/pasted from another page, I see there's a bunch of http links that should be https, so I've made a new issue for that: https://github.com/IATI/IATI-Dashboard/issues/736 (this should be easy, but I've put off doing it now to avoid conflicts with your current work).

simon-20 commented on iati_dashboard/ui/views.py:
```diff
@@ -543,6 +543,41 @@ def exploringdata_codelists_detail(request, major_version=None, attribute=None):
+    context["element"] = attribute
+    context["major_version"] = major_version
+    context["values"] = values
+    # FIXME this is duplicated above
```
Should these instances of `FIXME` be addressed as part of this? Or is this logged in a GH Issue somewhere?

Bjwebb replied:
I've resolved these in this PR now.

simon-20 commented on iati_dashboard/tests/test_validate_html.py:
```diff
@@ -0,0 +1,36 @@
+    if page.endswith(".json"):
+        pytest.skip()
+    # Skip those pages we know don't validate
+    # The plan is to fix them and then remove these
```
Is the task of fixing the pages with known validation issues logged somewhere?

Bjwebb replied:
I've added a link to an issue.

Bjwebb committed:
tests: Validate against W3C HTML validator

And, set this up on GitHub Actions.

https://github.com/IATI/IATI-Dashboard/issues/716

Bjwebb committed:
feat: Add codes per publisher per codelist page to replace popup

https://github.com/IATI/IATI-Dashboard/issues/653
